lib/galaxy/managers/tags.py

```diff
@@ -0,0 +1,55 @@
+from typing import (
+    List,
+    Optional,
+)
+
+from pydantic import (
+    BaseModel,
+    Field,
+)
+
+from galaxy.managers.context import ProvidesUserContext
+
+
+class ItemTagsPayload(BaseModel):
+    item_id: str = Field(
+        ...,  # This field is required
+        title="Item ID",
+        description="The `encoded identifier` of the item whose tags will be updated.",
+    )
+    item_class: str = Field(
+        ...,  # This field is required
+        title="Item class",
+        description="The name of the class of the item.",
```

Comment:
Can we make this an enum of all taggable item classes ?

Reply:
Done! Great suggestion!
I've added to the enum the name of the classes that are listed in [GalaxyTagHandler](https://github.com/galaxyproject/galaxy/blob/6e71c75d9b4f32cb85fa179df9bfd9b94da03f06/lib/galaxy/model/tags.py#L350) but I'm not sure if those are all of them or I may have missed some.

Reply:
That looks complete to me. I've been thinking about maybe using a metaclass that will register new taggable classes (or https://stackoverflow.com/a/50099920), since they should all be subclass of ItemTagAssociation https://github.com/mvdbeek/galaxy/blob/5d1d29d838983748704cc89a246799e2224b2abc/lib/galaxy/model/__init__.py#L6348

But a simple enum here is fine for now.

Reply:
Thanks for the hint!
I changed the Enum and now it is dynamically generated based on the names of the subclasses of `ItemTagAssociation`. This is a much better approach since we already found that there were some classes missing:
![image](https://user-images.githubusercontent.com/46503462/106461067-90127c00-6494-11eb-91ab-a5870589e0cc.png)

However, this brings some issues:
- Mypy can not correctly detect the Enum type and the only workaround to avoid the error that I found was ignoring it.
- The OpenAPI documentation does not show the enum values as it does with the regular Enums and the JSON that represents the model does not show the `item_class` property anymore. I'm not sure why :/